drivers/digital-io/max22200/iio_max22200.c

```diff
@@ -0,0 +1,1230 @@
+		return max22200_set_ch_hit(desc->max22200_desc,
+					   channel->address, hit_c,
+					   (uint8_t)val);
+	default:
+		return -EINVAL;
+	}
+}
+
+/**
+ * @brief Write the hold attribute for a specific channel.
+ * @param dev - The iio device structure.
+ * @param buf - Buffer to be filled with requested data.
+ * @param len - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv - Private descriptor
+ * @return 0 in case of success, error code otherwise
+ */
+static int max22200_iio_write_hold(void *dev, char *buf, uint32_t len,
+				   const struct iio_ch_info *channel,
+				   intptr_t priv)
+{
+	struct max22200_iio_desc *desc = dev;
+	uint32_t val;
+
+	iio_parse_value(buf, IIO_VAL_INT, (int32_t *)&val, NULL);
+
+	return max22200_set_ch_hold(desc->max22200_desc, channel->address,
+				    (uint8_t)val);
+}
+
+/**
+ * @brief Write the channel_trig attribute for a specific channel.
+ * @param dev - The iio device structure.
+ * @param buf - Buffer to be filled with requested data.
+ * @param len - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv - Private descriptor
+ * @return 0 in case of success, error code otherwise
+ */
+static int max22200_iio_write_channel_trig(void *dev, char *buf, uint32_t len,
+		const struct iio_ch_info *channel,
+		intptr_t priv)
+{
+	struct max22200_iio_desc *desc = dev;
+	enum max22200_trig ch_trig;
+	size_t i;
+	uint32_t val;
+
+	for (i = 0; i < NO_OS_ARRAY_SIZE(max22200_channel_trig_avail); i++) {
+		if (!strcmp(buf, max22200_channel_trig_avail[i])) {
+			ch_trig = i;
+			break;
+		}
+	}
+
+	if (i == NO_OS_ARRAY_SIZE(max22200_channel_trig_avail))
+		return -EINVAL;
+
+	val = no_os_field_prep(MAX22200_TRGNSP_IO_MASK, ch_trig);
+
+	return max22200_set_ch_trig(desc->max22200_desc, channel->address,
+				    val);
+}
+
+/**
+ * @brief Write the ch_mode attribute for a specific channel.
+ * @param dev - The iio device structure.
+ * @param buf - Buffer to be filled with requested data.
+ * @param len - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv - Private descriptor
+ * @return 0 in case of success, error code otherwise
+ */
+static int max22200_iio_write_ch_mode(void *dev, char *buf, uint32_t len,
+				      const struct iio_ch_info *channel,
+				      intptr_t priv)
+{
+	struct max22200_iio_desc *desc = dev;
+	enum max22200_ch_drive ch_drive;
+	enum max22200_ch_side ch_side;
+	enum max22200_ch_op_mode ch_op_mode;
+	size_t i;
+	int ret;
+
+	ret = max22200_get_ch_mode(desc->max22200_desc, channel->address,
+				   &ch_drive, &ch_side, &ch_op_mode);
+	if (ret)
+		return ret;
+
+	switch (priv) {
+	case MAX22200_IIO_CH_DRIVE:
+		for (i = 0; i < NO_OS_ARRAY_SIZE(max22200_ch_drive_avail); i++) {
+			if (!strcmp(buf, max22200_ch_drive_avail[i])) {
+				ch_drive = i;
+				break;
+			}
+		}
+
+		if (i == NO_OS_ARRAY_SIZE(max22200_ch_drive_avail))
```

Comment:
all these checks can be dropped if you return directly in the for loop. same applies in other functions of the driver. This is just a suggestion.

Reply:
commited.